src/python/dxpy/workflow_builder.py

```diff
@@ -325,53 +364,63 @@ def _get_validated_json_for_build_or_update(json_spec, args):
     """
     validated = copy.deepcopy(json_spec)
 
-    dxpy.executable_builder.inline_documentation_files(validated, args.src_dir)
+    if not args._from:
+        dxpy.executable_builder.inline_documentation_files(validated, args.src_dir)
 
     if 'title' not in json_spec:
-        logger.warn("dxworkflow.json is missing a title, please add one in the 'title' field")
+        logger.warn("workflow spec is missing a title, please add one in the 'title' field")
 
     if 'summary' not in json_spec:
-        logger.warn("dxworkflow.json is missing a summary, please add one in the 'summary' field")
+        logger.warn("workflow spec is missing a summary, please add one in the 'summary' field")
     else:
         if json_spec['summary'].endswith('.'):
             logger.warn("summary {} should be a short phrase not ending in a period".format(json_spec['summary'],))
 
     return validated
 
 
-def _assert_executable_regions_match(workflow_enabled_regions, workflow_spec):
+def _assert_executable_regions_match(workflow_enabled_regions: set, workflow_spec):
     """
     Check if the global workflow regions and the regions of stages (apps) match.
-    If the workflow contains any applets, the workflow can be currently enabled
+    If the workflow contains any applets, then the workflow can be currently enabled
     in only one region - the region in which the applets are stored.
     """
+    if not workflow_enabled_regions: # empty set
+        return workflow_enabled_regions
+    
     executables = [i.get("executable") for i in workflow_spec.get("stages")]
 
     for exect in executables:
-
-        if exect.startswith("applet-") and len(workflow_enabled_regions) > 1:
-            raise WorkflowBuilderException("Building a global workflow with applets in more than one region is not yet supported.")
+        if exect.startswith("applet-"):
+            applet_region = dxpy.api.applet_describe(dxpy.api.applet_describe(exect)["project"])["region"]
```

Comment:
How about using dx bindings here:
```suggestion
            applet_project = dxpy.DXApplet(exect).project
            applet_region = dxpy.DXProject(applet_project).region
```